File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/ProductBridge.kt
package com.superwall.superwallkit_flutter.bridges
import com.superwall.sdk.models.product.Product
import com.superwall.sdk.models.product.ProductType

fun Product.toJson(): Map<String, Any?> {
    return mapOf(
        "type" to type.toJson(),
        "id" to id
    )
}

fun ProductType.toJson(): String {
    return when (this) {
        ProductType.PRIMARY -> "primary"
        ProductType.SECONDARY -> "secondary"
        ProductType.TERTIARY -> "tertiary"
    }
}

File: ios/Classes/Bridges/ProductBridge.swift
import Flutter
import SuperwallKit

extension Product {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["type"] = type.toJson()
    dict["id"] = id
    return dict
  }
}

extension ProductType {
  func toJson() -> String {
    switch self {
      case .primary:
        return "primary"
      case .secondary:
        return "secondary"
      case .tertiary:
        return "tertiary"
    }
  }
}
